### DISCUSSION TITLE
Why kitty does not render italics with a bad font when other terminals do?

### DISCUSSION THREAD

**Discussion by nativerv** [kovidgoyal/kitty#7593] [General]
For the longest time i thought kitty just didn't have italics support for some reason. I don't know, a troll from the dev or something. I was not using kitty as my main terminal so i didn't care enough to investigate further than just looking at the output of the according echo command.

Now after googling it i understand that this is a font issue. In the past i've used mostly Fira Code Nerd Font, currently i'm using `DejaVuBQNSansMono.ttf` which i've got from somewhere from the BQN project. And neither of them have italics in kitty. And you actually [recommend](https://github.com/kovidgoyal/kitty/issues/523#issuecomment-386849914) DejaVu Sans as a sanity check, which indeed enabled italics in kitty when i removed the BQN glyph fork and it used the original one.

But the thing is that literally any other terminal that i try renders italics anyway by whatever means, just forces it or something.
St, foot, urxvt, alactritty all render italics with the current font and i'm pretty sure they did with Fira Code. Only xterm don't, which i'm not sure if it is a font issue or xterm issue.

And then the issue that i referenced above actually had bad font but it still rendered italics in kitty, just broken ones.

So, what's going on? Is that expected behavior or a my system issue? If the former, is there some established workflow for patching custom (or not so custom) fonts with proper italics?

**Comment by kovidgoyal** [comment]
If your font does not have an italic face, as Fira Code for example
doesnt, you wont get italics. kitty's automatic font fallback will
never choose a font from another family as these often dont have
compatible metrics. If you want to use a font without an italic face but
still have italics, set italic_font in kitty.conf to some other face. But
be warned its quite likely the italic characters will not fit in the
cell and kitty unlike other terminal emulators is a strict grid based
display, so you will get either chopped off or resized and weird looking
italic glyphs for some characters.

**Reply by nativerv** [reply]
> If you want to use a font without an italic face but
still have italics, set italic_font in kitty.conf to some other face

But can i set that in fontconfig somehow so that it's only one source of truth for everything on my desktop?
